sources/packages/api/src/services/institution-location/institution-location.service.ts

```diff
@@ -217,4 +217,38 @@ export class InstitutionLocationService extends RecordDataModelService<Instituti
       .where("location.id = :locationId", { locationId })
       .getRawOne();
   }
+
+  /**
+   * Get an institution's locations with designation statuses.
+   * @param institutionId institution id
+   * @returns InstitutionLocation[]
+   */
+  async getInstitutionLocationsWithDesignationStatus(
```

Comment:
The type being returned here does not seem to represent the type actually returned by the `.getRawMany()`.
Please ensure that interface returned has all the expected properties and do the proper mapping between the `.getRawMany()` and the object returned by the method.